src/ui/base-page-storyEdit.js

```diff
@@ -370,24 +370,35 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/main/LICENSE.
     };
 
     /**
-     * Loads story content from a given key in the browser's localStorage object.
-     * If a story was successfully loaded, then the current story is updated with
-     * this previously-saved data. If no story is loaded, then a new one is saved
-     * to the server.
+     * Loads an existing story from the server if a story id is provided as a query
+     * parameter. Otherwise loads story content from a given key in the browser's
+     * localStorage object. If a story was successfully loaded, then the current
+     * story is updated with this previously-saved data. If no story is loaded,
+     * then a new one is saved to the server.
      *
      * @param {String} storyAutosaveKey - the key to load the story content from
      * @param {Component} storyEdit - an instance of `sjrk.storyTelling.base.page.storyEdit`
      */
     sjrk.storyTelling.base.page.storyEdit.initializeStory = function (storyAutosaveKey, storyEdit) {
+
         try {
-            // localStorage can only store string values
+            /*
+            * load from existing story model if available
```

Comment:
Thanks for documenting this list of scenarios. It's helpful to understand the situation.

But it looks overwhelming by covering these different combinations to guess what story data should be loaded.

I wonder if this can be simplified by passing in an "edit"/"create" state as a model value. Then the logic would be: 1. at the "create" state, load saved story data from the local storage; 2. at "edit", load data retrieved from the server.

Reply:
I'm not sure that's possible. In particular for the case where you are editing a published story and also want to make use of the auto save local storage. In the implementation I've currently included, it will use the local storage if that's available with the same id as the incoming changes from the server.